src/App.js

```diff
@@ -29,22 +29,30 @@ class App extends Component {
   componentDidMount() {
     this.tokenMatch();
   }
+  
+  editRouteHandler() {
+    let editRoute = null;
+    if (localStorage.getItem('user_Id') !== null && localStorage.getItem('user_Id') === localStorage.getItem('post_user_Id')){
+      editRoute = <Route path="/post/:post_id/edit" exact component={EditPost} />
+    } else{
+      editRoute = <Route path="/post/:post_id/edit" component={NoMatch} />
+    }
+    return editRoute
```

Comment:
This seems it can be done using just ternary operators. So...

```
localStorage.getItem('user_Id') !== null && localStorage.getItem('user_Id') === localStorage.getItem('post_user_Id')
  ? <Route path="/post/:post_id/edit" exact component={EditPost} />
  : <Route path="/post/:post_id/edit" component={NoMatch} />
```

Ternary operators already return regardless. No need to declare a `editRoute`.

Reply:
oh shoot thats tite and  good to know !